common/src/main/java/org/openshift/ping/common/stream/TokenStreamProvider.java

```
@@ -0,0 +1,101 @@
+            httpsConnection.setSSLSocketFactory(factory);
+            if (log.isLoggable(Level.FINE)) {
+                log.fine(String.format("Using HttpsURLConnection with SSLSocketFactory [%s] for url [%s].", factory, url));
+            }
+        } else {
+            if (log.isLoggable(Level.FINE)) {
+                log.fine(String.format("Using URLConnection for url [%s].", url));
+            }
+        }
+        
+		if (token != null) {
+            // curl -k -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" \
+            // https://172.30.0.2:443/api/v1/namespaces/dward/pods?labelSelector=application%3Deap-app
+            headers.put("Authorization", "Bearer " + token);
+        }
+		return connection.getInputStream();
+	}
+	
+	private TrustManager[] configureCaCert(String caCertFile) throws Exception {
+        if (caCertFile != null) {
+            try {
+                InputStream pemInputStream = openFile(caCertFile);
+                CertificateFactory certFactory = CertificateFactory.getInstance("X509");
+                X509Certificate cert = (X509Certificate)certFactory.generateCertificate(pemInputStream);
+    
+                KeyStore trustStore = KeyStore.getInstance("JKS");
+                trustStore.load(null);
+    
+                String alias = cert.getSubjectX500Principal().getName();
+                trustStore.setCertificateEntry(alias, cert);
+    
+                TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+                trustManagerFactory.init(trustStore);
+    
+                return trustManagerFactory.getTrustManagers();
+            } catch (FileNotFoundException fnfe) {
+                log.log(Level.WARNING, "ca cert file not found.");
+                return InsecureStreamProvider.INSECURE_TRUST_MANAGERS;
```

IMHO, if the file is missing, we should throw an exception. Right now, people may not notice the warning and may end up talking to an imposter.